src/lsqecc/pauli_rotations/circuit.py

```diff
@@ -321,6 +321,14 @@ def load_from_pyzx(circuit) -> "PauliOpCircuit":
         print("Gate Missed: ", gate_missed)
         return ret_circ
 
+    @staticmethod
+    def from_list(pauli_op_list: List[PauliProductOperation]):
+        c = PauliOpCircuit(pauli_op_list[0].qubit_num)
+        for op in pauli_op_list:
+            assert c.qubit_num == op.qubit_num
```

Comment:
Minor Suggestion: I think we should call it `num_qubits` instead `qubit_num`, because later might thought of as returning the index. There is scope for ambiguity.

Reply:
yep we definitely should ... I think this PR is already big enough though